rosette/src/Import.cc

```diff
@@ -19,12 +20,188 @@
 #include <string>
 #include <map>
 
+extern pOb NILmeta;
+extern CompoundPattern* NILpattern;
 
 ObjectCodePB::ObjectCode importObjectCode;
+typedef struct {
+    IdType id;
+    int pbIndex;
+    pOb object;
+} ObjectInfo;
+
+std::map<IdType, ObjectInfo> objects;
 
 pOb createRosetteObject(const ObjectCodePB::Object & ob) {
 
+    pOb retval = NULL;
+
     switch(ob.type()) {
+    case ObjectCodePB::OT_ACTOR:
+        retval = Actor::create();
+        break;
+    case ObjectCodePB::OT_BLOCK_EXPR:
+        retval = BlockExpr::create(NULL);
+        break;
+    case ObjectCodePB::OT_CHAR:
+        retval = RBLCHAR(ob.char_().value());
+        break;
+    case ObjectCodePB::OT_CODE:
+        // Handled separately
+        break;
+    case ObjectCodePB::OT_EXPANDED_LOCATION:
+        // ???
+        break;
+    case ObjectCodePB::OT_FIXNUM:
+        retval = FIXNUM(ob.fixnum().value());
+        break;
+    case ObjectCodePB::OT_FLOAT:
+        retval = Float::create((double)ob.float_().value());
+        break;
+    case ObjectCodePB::OT_FREE_EXPR:
+        retval = FreeExpr::create(NULL, NULL);
+        break;
+    case ObjectCodePB::OT_GOTO_EXPR:
+        retval = GotoExpr::create(SYMBOL(ob.goto_expr().label().symbol().name().c_str()));
+        break;
+    case ObjectCodePB::OT_IF_EXPR:
+        retval = IfExpr::create(NULL, NULL);
+        break;
+    case ObjectCodePB::OT_LABEL_EXPR:
+        retval = LabelExpr::create(SYMBOL(ob.goto_expr().label().symbol().name().c_str()), NULL);
+        break;
+    case ObjectCodePB::OT_LET_EXPR:
```

Comment:
All the ones with `NULL` are todos?

Reply:
Yes.  Those are ones to be filled in during the link stage.